lib/Map/DataTable.js

```diff
@@ -288,26 +332,71 @@ DataTable.prototype.getCsvTable = function () {
 };
 
 /**
-* Set the current data variable
+* Set the current data variable.
 *
-* @param {String} varName The name of the variable
+* @param {String} varName The variable name which has been toggled.
+* @param {Boolean} [makeActive=True] Has this been made active or inactive? Note if allowMultiple is True, this is ignored.
 *
 */
-DataTable.prototype.setDataVariable = function (varName) {
-    if (!defined(this.variables[varName])) {
-        varName = undefined;
+DataTable.prototype.setDataVariable = function (varName, makeActive) {
+    // Note we take care to ensure selectedNames is a new object so it triggers knockout's tracking.
+    if (defined(this.variables[varName])) {
+        if (this.allowMultiple) {
+            // If it's not in the list, add it; if it is, remove it.
+            var index = this.selectedNames.indexOf(varName);
+            if (index === -1) {
+                this.selectedNames = this.selectedNames.concat([varName]);  // use concat not push to clone the array.
+            } else {
+                this.selectedNames.splice(index, 1);
+                this.selectedNames = this.selectedNames.slice();  // slice to clone the array.
+            }
+        } else {
+            // If this don't allow multiple selections, just reset the list to the new selection.
+            if (makeActive === false) {
+                this.selectedNames = [];
+            } else {
+                this.selectedNames = [varName];
+            }
+        }
+    } else {
+        this.selectedNames = [];
+    }
+};
+
+/**
+ * Set the current data variable to a reasonable default. Return the data variable.
+ * @returns {DataVariable} The chosen data variable.
+ */
+DataTable.prototype.setDefaultDataVariable = function() {
```

Comment:
Should this be `assignDefaultDataVariable` or `chooseDefaultDataVariable`? Convention has `setX` takes an argument, no?

Also we should make this choice smarter - picking the leftmost column often means we choose a name field, which works out badly.